captum/optim/models/_common.py

```diff
@@ -21,9 +21,13 @@ def get_layers(net: nn.Module, prefix: List = []) -> None:
             for name, layer in net._modules.items():
                 if layer is None:
                     continue
-                separator = "" if str(name).isdigit() else "."
-                name = "[" + str(name) + "]" if str(name).isdigit() else name
-                layers.append(separator.join(prefix + [name]))
+                name_str = "".join(
+                    [
+                        ("[" + str(i) + "]" if str(i).isdigit() else "." + i)
+                        for i in prefix + [name]
+                    ]
+                ).strip()
+                layers.append(name_str[1:] if name_str[0] == "." else name_str)

```

Comment:
do you think we can use the string `name_str` as the `prefix` argument? so no need to repeat the `list -> str` conversion for the prefix part in each recursive call.

Reply:
I don't think that we can, as the naming requires knowing the name of the module that the current module is inside.

Reply:
I mean sth like this, where the `prefix` contains the chain of modules
```py
def get_layers(net: nn.Module, prefix:str = "") -> None:
  for name, layer in net._modules.items():
    delimiter = "." if prefix else ""
    name_str = f"{prefix}[{name}]" if str(name).isdigit() else f"{prefix}{delimiter}{name}"
    layer_append(name_str)
    get_layers(layer, prefix=name_str)
```

`prefix` sounds more naturally as a variable for string and seems we do not need to save the list. Let me know if I miss sth

Reply:
@aobo-y Ah, I see, and your code suggestion looks good! I've replaced the existing code with it.